shared/js/initApp.js

```diff
@@ -150,7 +150,35 @@ $(window).on('function_editor_closed', function() {
   }
 })(appOptions.level);
 
+dashboard.saveProject = function(callback) {
+  var app_id = dashboard.currentApp.id;
+  dashboard.currentApp.startBlocks = Blockly.Xml.domToText(Blockly.Xml.blockSpaceToDom(Blockly.mainBlockSpace));
+  if (app_id) {
+    storageApps().update(app_id, dashboard.currentApp, function(data) {
+      callback(data);
+    });
+  } else {
+    storageApps().create(dashboard.currentApp, function(data) {
+      if (history) {
+        history.pushState({}, '', '?id=' + data.id);
```

Comment:
No need to fix this now... but this looks like it needs a router.

I wonder if it would be possible to codegen a client side router from the rails routes, writing this twice every time will start to get annoying...